### PR TITLE
Feature/gfi-json-display

### PR DESCRIPTION
**Pull request by JoshGx2000**
If the info format is "unavailable" for the layers when attempting to get the feature info, then the GetFeatureInfo request is skipped – a popup saying that layer data is currently unavailable is shown instead.

New popup also used in the event that the JSON popup has no data to display.

### CODE REVIEW

File: frontend/src/cljs/imas_seamap/map/utils.cljs [hunk 1]
```diff
@@ -226,14 +234,16 @@
...
+        property-rows (str/join "" (map (fn [property] (property-to-row property)) properties))
+        properties []
+        id nil]
     (if (or id (not-empty properties))
       {:info (str
               "<div class=\"feature-info-json\">"
               "<h4>" id "</h4>"
               "<table>" property-rows "</table>"
               "</div>")}
-      {:status :feature-info/empty})))
```
Comment (markhepburn) [inline frontend/src/cljs/imas_seamap/map/utils.cljs:236]: This worries me a bit -- is this saying that there'll never be no results returned?  Even if you click in the ocean, or on land?

File: frontend/src/cljs/imas_seamap/map/events.cljs [hunk 2]
```diff
@@ -31,50 +31,61 @@
...
+                                   :response-priority 99
+                                   :had-insecure?     had-insecure?
+                                   :candidate         nil}
+                   :feature       {:status   :feature-info/waiting
+                                   :location point})}
+        
+        info-format
+        (assoc
+         :http-xhrio
+         (when info-format
```
Comment (markhepburn) [inline frontend/src/cljs/imas_seamap/map/events.cljs:53]: This looks redundant -- isn't the cond-> clause "info-format" already implicitly a "truthy" condition?

File: frontend/src/cljs/imas_seamap/map/events.cljs [hunk 3]
```diff
@@ -31,50 +31,61 @@
         ;; Note, top layer, last in the list, must be first in our search string:
         layers->str   #(->> % (map layer-name) reverse (string/join ","))
         request-id    (gensym)
-        had-insecure? (->> db :map :active-layers (some #(is-insecure? (:server_url %))))]
+        had-insecure? (->> db :map :active-layers (some #(is-insecure? (:server_url %))))
+        info-format (get-layers-info-format active-layers)]
```
Comment (markhepburn) [inline frontend/src/cljs/imas_seamap/map/events.cljs:35]: Minor, but might as well align these

File: frontend/src/cljs/imas_seamap/map/events.cljs (hunk 2 above)
Comment (markhepburn) [reply frontend/src/cljs/imas_seamap/map/events.cljs:53]: In fact looking further, I think this would be better extracted as a function at this point -- you're using cond-> to implement if-then-else, and the whole lot is getting a bit hard to follow.

**Review by markhepburn — CHANGES_REQUESTED** [review]
Mind having a bit more of a tidy please?  There's a couple of things, both stylistic and possibly a logic / semantic issue (handling "normal" no results).

Also I'm about to push, so it may not merge cleanly... sorry :)

File: frontend/src/cljs/imas_seamap/map/utils.cljs (hunk 1 above)
Comment (JoshGx2000) [reply frontend/src/cljs/imas_seamap/map/utils.cljs:236]: You're absolutely right. Reversing that change now.

File: frontend/src/cljs/imas_seamap/map/events.cljs (hunk 2 above)
Comment (JoshGx2000) [reply frontend/src/cljs/imas_seamap/map/events.cljs:53]: For your first comment, yep, that's silly me who forgot to delete some code I made when trying out solutions. For your second comment, I'll work on that now.

File: frontend/src/cljs/imas_seamap/map/events.cljs [hunk 4]
```diff
@@ -25,61 +25,68 @@
...
+      (let [params {:REQUEST       "GetFeatureInfo"
+                    :LAYERS        layers
+                    :QUERY_LAYERS  layers
+                    :WIDTH         (:x size)
+                    :HEIGHT        (:y size)
+                    :BBOX          (bounds->str bounds)
+                    :FEATURE_COUNT 5
+                    :STYLES        ""
+                    :X             x
+                    :Y             y
```
Comment (markhepburn) [inline frontend/src/cljs/imas_seamap/map/events.cljs:49]: Final request please -- would you mind merging develop into your branch?  I made some concurrent edits that will be clobbered by these lines otherwise (essentially, x/y/width/height is fixed, and bounds is calculated separately from the viewport)

**Review by markhepburn — CHANGES_REQUESTED** [review]
Looks good thanks -- there's a merge conflict you'll need to take care of first though (sorry), I'm suggesting merging develop to resolve it in this branch before we merge the PR.
